flink-clients/src/main/java/org/apache/flink/client/deployment/ClusterSpecification.java

```diff
@@ -18,20 +18,35 @@
 
 package org.apache.flink.client.deployment;
 
+import org.apache.flink.runtime.jobmanager.JobManagerProcessSpec;
+import org.apache.flink.util.Preconditions;
+
+import static org.apache.flink.runtime.jobmanager.JobManagerProcessUtils.createDefaultJobManagerProcessSpec;
+
 /**
  * Description of the cluster to start by the {@link ClusterDescriptor}.
  */
 public final class ClusterSpecification {
+	private final JobManagerProcessSpec masterProcessSpec;
```

Comment:
ok, I change it know to not extend `ClusterSpecification`. We can continue discussion in FLINK-16734. In general, I think it would be good to derive `JobManagerProcessSpec` in one place for all integrations and have plan for this if it is not in `ClusterSpecification`.

Reply:
True, it makes sense to derive `JobManagerProcessSpec` in one place for all integrations.